### PR TITLE
Add Printer Exporter

### PR DESCRIPTION
**Pull request by soheilrt**
Addresses: https://github.com/prometheus-community/windows_exporter/issues/1470

This PR Adds the exporter for windows, it uses WMI classes (Win32_Printer and Win32_PrintJob) to fetch the data and publish them to prometheus.

References:
* https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/win32-printer
* https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/win32-printjob

An example of the output is like below: 
```
# HELP windows_printer_job_count Number of jobs processed by the printer since the last reset
# TYPE windows_printer_job_count counter
windows_printer_job_count{printer="BE_TEST_PT01"} 4
windows_printer_job_count{printer="MO_TEST_PT01"} 0
windows_printer_job_count{printer="Microsoft Print to PDF"} 0
windows_printer_job_count{printer="Microsoft XPS Document Writer"} 0
# HELP windows_printer_job_status A counter of printer jobs by status
# TYPE windows_printer_job_status gauge
windows_printer_job_status{printer="BE_TEST_PT01",status="Error"} 1
windows_printer_job_status{printer="BE_TEST_PT01",status="UNKNOWN"} 3
# HELP windows_printer_status Printer status
# TYPE windows_printer_status gauge
windows_printer_status{printer="BE_TEST_PT01",status="Other"} 1
windows_printer_status{printer="MO_TEST_PT01",status="Idle"} 1
windows_printer_status{printer="Microsoft Print to PDF",status="Idle"} 1
windows_printer_status{printer="Microsoft XPS Document Writer",status="Idle"} 1
```

Checklist:
- [x]  Add Documentations
- [x] Manual Testing
- [x] Adding to benchmark test

### CODE REVIEW

**Comment by soheilrt** [comment]
Hi @prometheus-community/windows_exporter-reviewers, Can I have an initial review on this please?

File: pkg/collector/config.go [hunk 1]
```diff
@@ -111,6 +112,7 @@ type Config struct {
 	Time                           time.Config                            `yaml:"time"`
 	Vmware                         vmware.Config                          `yaml:"vmware"`
 	VmwareBlast                    vmware_blast.Config                    `yaml:"vmware_blast"`
+	Printer                        printer.Config                         `yaml:"printer"`
```
Comment (jkroepke) [inline pkg/collector/config.go:115]: Could you please put printer in order?

File: pkg/collector/config.go [hunk 2]
```diff
@@ -169,4 +171,5 @@ var ConfigDefaults = Config{
 	Time:                           time.ConfigDefaults,
 	Vmware:                         vmware.ConfigDefaults,
 	VmwareBlast:                    vmware_blast.ConfigDefaults,
+	Printer:                        printer.ConfigDefaults,
```
Comment (jkroepke) [inline pkg/collector/config.go:174]: Could you please put printer in order?

File: pkg/collector/collector.go [hunk 3]
```diff
@@ -140,7 +141,7 @@ func NewWithConfig(logger log.Logger, config Config) Collectors {
 	collectors[time.Name] = time.New(logger, &config.Time)
 	collectors[vmware.Name] = vmware.New(logger, &config.Vmware)
 	collectors[vmware_blast.Name] = vmware_blast.New(logger, &config.VmwareBlast)
-
```
Comment (jkroepke) [inline pkg/collector/collector.go:143]: Could you please put printer in order?

File: pkg/collector/map.go [hunk 4]
```diff
@@ -115,6 +116,7 @@ var Map = map[string]types.CollectorBuilderWithFlags{
 	time.Name:                            time.NewWithFlags,
 	vmware.Name:                          vmware.NewWithFlags,
 	vmware_blast.Name:                    vmware_blast.NewWithFlags,
+	printer.Name:                         printer.NewWithFlags,
```
Comment (jkroepke) [inline pkg/collector/map.go:119]: Could you please put printer in order?

**Review by jkroepke — CHANGES_REQUESTED** [review]
it's an initial review yet. I have to test it on my machine.

**Comment by soheilrt** [comment]
Hi @jkroepke, Would it be possible to approve workflows for this PR, I don't have a windows machine and it's a bit pain to test my changes every time on a virtual machine.

**Comment by soheilrt** [comment]
Hi again @jkroepke , I guess PR is ready for a review, I would appreciate it if you could have a look at it when you have time.


Also, Could you please share the release schedule?

**Comment by soheilrt** [comment]
Hi @jkroepke, did you have time to have a look at my PR? It's a bit crucial for us to have these metrics. I would really appreciate your help on this.

File: pkg/collector/printer/printer.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+		ch <- prometheus.MustNewConstMetric(
+			c.printerStatus,
+			prometheus.GaugeValue,
+			1,
+			printer.Name,
+			printerStatus,
+		)
+		ch <- prometheus.MustNewConstMetric(
+			c.printerJobCount,
+			prometheus.GaugeValue,
```
Comment (jkroepke) [inline pkg/collector/printer/printer.go:174]: JobCounter is expect to increase only? If yes, it should be a Counter.

**Review by jkroepke — COMMENTED** [review]
tested on local system, seems working for me. I have one question left.

File: pkg/collector/printer/printer.go (hunk 5 above)
Comment (jkroepke) [reply pkg/collector/printer/printer.go:174]: good link!

According the linked docs, the value is documented as counter, too.

Comment (soheilrt) [reply pkg/collector/printer/printer.go:174]: Sure, I'll adjust it.

Comment (soheilrt) [reply pkg/collector/printer/printer.go]: Done ✅

**Review by jkroepke — APPROVED** [review]
Thanks

**Pull request merged**